src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/InterOp/HwndKeyboardInputProvider.cs

```diff
@@ -117,6 +117,13 @@ bool IKeyboardInputProvider.AcquireFocus(bool checkOnly)
                         // window handle for windows attached to the calling
                         // thread's queue.
                         //
+                        int thisPid = 0;
+                        int childPid = 0;
+                        UnsafeNativeMethods.GetWindowThreadProcessId(new HandleRef(null, focus), out childPid);
+                        UnsafeNativeMethods.GetWindowThreadProcessId(thisWindow, out thisPid);
+
+                        if (childPid != thisPid)
+                            UnsafeNativeMethods.TrySetFocus(thisWindow);
```

Comment:
This is exactly what the comment at line 85 says we should _not_ do.  Which suggests that this will break something (even though it fixes WebView2).  This is not the right fix.

Lines 110-118 explain how VS participates in cross-thread focus behavior.  It sounds like WebView2 is doing some, but not all, of that work:  they are (presumably) calling AttachThreadInput so that GetFocus returns their window, but not reposting kb messages to our thread.   So the right fix might belong to WebView2.

Reply:
@SamBent The comments also says, "This is typically for the menu Popup". Other than Menu-popup, should current control always grab the focus, if AcquireFocus is called?
For multi-process application, where we cannot trust the implementation by others to do the right thing, this may become appropriate to force move the focus, if child-process isn't behaving the way we expect it to.